src/test/java/com/mongodb/hibernate/jdbc/MongoConnectionProviderTests.java

```diff
@@ -35,72 +35,64 @@
 import com.mongodb.event.ClusterListener;
 import com.mongodb.event.ClusterOpeningEvent;
 import com.mongodb.hibernate.BuildConfig;
-import com.mongodb.hibernate.service.MongoClientCustomizer;
+import com.mongodb.hibernate.service.MongoDialectConfigurator;
 import java.util.function.Consumer;
+import org.hibernate.HibernateException;
 import org.hibernate.SessionFactory;
 import org.hibernate.boot.MetadataSources;
 import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
-import org.hibernate.cfg.JdbcSettings;
+import org.hibernate.cfg.AvailableSettings;
 import org.hibernate.engine.jdbc.connections.spi.ConnectionProvider;
 import org.hibernate.engine.spi.SessionFactoryImplementor;
 import org.hibernate.service.spi.ServiceException;
 import org.jspecify.annotations.Nullable;
 import org.junit.jupiter.api.Nested;
 import org.junit.jupiter.api.Test;
-import org.junit.jupiter.params.ParameterizedTest;
-import org.junit.jupiter.params.provider.ValueSource;
 
 class MongoConnectionProviderTests {
 
     @Nested
-    class MongoClientCustomizerTests {
+    class MongoDialectConfiguratorTests {
 
-        @ParameterizedTest
-        @ValueSource(booleans = {true, false})
-        void testMongoClientCustomizerTakeEffect(boolean customizerAppliesConnectionString) {
```

Comment:
`customizerAppliesConnectionString` is not longer relevant, as we apply the connection string before using the configurator. See https://github.com/mongodb/mongo-hibernate/pull/31#discussion_r1939903441.